### PR TITLE
Add Date type assertion

### LINKED ISSUE #94: Date assertion
It would be nice to have a date assertion that worked like the following:

```js
(new Date()).should.be.a.Date();
```

Is there any specific reason to not support it? Would you guys accept a PR for it?

Thanks.

### PR DESCRIPTION
**Pull request by ruiquelhas**
Resolves #94.

### CODE REVIEW

File: test/ext/type.test.js [hunk 1]
```diff
@@ -224,4 +226,19 @@ describe('type', function() {
     }, 'expected Array [] to be an error',
        '    expected Array [] to be an instance of Error');
   });
+
+  it.only('test Date', function() {
```
Comment (fixe) [inline test/ext/type.test.js]: Remove `.only`.

File: test/ext/type.test.js [hunk 2]
```diff
@@ -224,4 +226,19 @@ describe('type', function() {
     }, 'expected Array [] to be an error',
        '    expected Array [] to be an instance of Error');
   });
+
+  it('test Date', function() {
+    (new Date()).should.be.a.Date();
+    (new Date(Date.now())).should.be.a.Date();
+    ('2016-01-06T12:22:03.269Z').should.be.a.Date();
```
Comment (btd) [inline test/ext/type.test.js]: It should not pass on strings, even on specially formatted.

File: test/ext/type.test.js [hunk 3]
```diff
@@ -224,4 +226,19 @@ describe('type', function() {
...
+    ('2015-02-27').should.be.a.Date();
+    // but
+    should(new Date()).should.not.be.a.Date();
```
Comment (btd) [inline test/ext/type.test.js]: That is pretty useless. `should(new Date())` returns `Assertion`, next you call `.should` on Assertion

File: test/ext/type.test.js [hunk 4]
```diff
@@ -224,4 +226,19 @@ describe('type', function() {
...
+
+    (new Date('foo')).should.not.be.a.Date();
```
Comment (btd) [inline test/ext/type.test.js]: It should be if `new Date('foo')` returns Date instance.

**Comment by btd** [comment]
Type assertion should only check if this.obj is Date, not that it is String or valid date.

**Comment by ruiquelhas** [comment]
Ok, I was aiming for a failure with `Invalid Date`, but I'll do as you suggest.

**Comment by btd** [comment]
Yep, i understand. It is better to add second method, like validDate, where you check it is Date and your check `isNaN(this.obj.getTime())`.

**Comment by btd** [comment]
I will publish new version at sunday, ok?

**Comment by ruiquelhas** [comment]
Great. Would you fancy a different PR for that second method, or do you feel it fits better as a custom user-land assertion?

**Comment by btd** [comment]
I am :+1: on second PR

**Pull request merged**